### PR TITLE
MAINT: update minimum NumPy version to 1.16.5 in a couple more places

### PR DESCRIPTION
**Pull request by rgommers**
Follow-up to gh-12846, which wasn't quite complete.

### CODE REVIEW

File: doc/source/toolchain.rst [hunk 1]
```diff
@@ -71,9 +71,10 @@ The table shows the NumPy versions suitable for each major Python version
 =================  ========================    ===========================
 2.7 (SciPy 1.2)      1.8.2                      1.16.x
 3.5 (SciPy 1.4)      1.13.3                     1.18.x
-3.6                  1.14.5                     >= 1.18.x
-3.7                  1.14.5                     >= 1.18.x
+3.6 (SciPy 1.5)      1.14.5                     >= 1.18.x
```
Comment (h-vetinari) [inline doc/source/toolchain.rst:74]:
```suggestion
3.6 (SciPy 1.5)      1.14.5                     1.19.x
```

File: doc/source/toolchain.rst [hunk 2]
```diff
@@ -71,9 +71,10 @@ The table shows the NumPy versions suitable for each major Python version
...
+3.7                  1.16.5                     >= 1.18.x
 3.8                  1.17.3                     >= 1.18.x
+3.9                  1.19.3                     >= 1.18.x
```
Comment (h-vetinari) [inline doc/source/toolchain.rst:77]: The maximum versions here could also uniformly move to `>= 1.20.x`...

Comment (h-vetinari) [reply doc/source/toolchain.rst:77]: Further down under `C Compilers`, python 3.9 should also be added (not necessarily in this PR)

File: doc/release/1.6.0-notes.rst [hunk 3]
```diff
@@ -17,9 +17,9 @@ run your code with ``python -Wd`` and check for ``DeprecationWarning`` s).
 Our development attention will now shift to bug-fix releases on the
 1.3.x branch, and on adding new features on the master branch.
 
-This release requires Python 3.6+ and NumPy 1.14.5 or greater.
+This release requires Python 3.6+ and NumPy 1.16.5 or greater.
```
Comment (tylerjereddy) [inline doc/release/1.6.0-notes.rst:20]: I guess 3.7+ now too, although maybe we can just keep this MR focused on its original purpose.

File: doc/source/toolchain.rst (hunk 1 above)
Comment (tylerjereddy) [reply doc/source/toolchain.rst:74]: @rgommers You ok with this modification?

File: doc/source/toolchain.rst (hunk 2 above)
Comment (h-vetinari) [reply doc/source/toolchain.rst:77]:
```suggestion
3.7                  1.16.5                     >= 1.20.x
3.8                  1.17.3                     >= 1.20.x
3.9                  1.19.3                     >= 1.20.x
```

Suggestion for updated numpy maximum versions for those python versions still supported by 1.20.

File: doc/release/1.6.0-notes.rst (hunk 3 above)
Comment (rgommers) [reply doc/release/1.6.0-notes.rst:20]: yeah I considered that and left it out on purpose, to not do two things at once.

File: doc/source/toolchain.rst (hunk 2 above)
Comment (rgommers) [reply doc/source/toolchain.rst]: makes sense, adopted

File: doc/source/toolchain.rst (hunk 1 above)
Comment (rgommers) [reply doc/source/toolchain.rst]: thanks, adopted

**Comment by rgommers** [comment]
Let's get this in straight away, and review the follow-up PR on `toolchain.rst` as well. Thanks for the reviews @h-vetinari and @tylerjereddy

**Pull request merged**